app/assets/javascripts/controllers/network_router/network_router_form_controller.js

```diff
@@ -65,47 +104,24 @@ ManageIQ.angular.app.controller('networkRouterFormController', ['$http', '$scope
     miqService.miqFlash("warn", "All changes have been reset");
   };
 
-  vm.filterNetworkManagerChanged = function() {
+  vm.filterNetworkManagerChanged = function(id) {
     miqService.sparkleOn();
-    $http.get('/network_router/network_router_networks_by_ems/' + vm.networkRouterModel.ems_id)
-      .then(getNetworkRouterFormByEmsData)
-      .catch(miqService.handleFailure);
-
-    miqService.getProviderTenants(function(data) {
-      vm.available_tenants = data.resources;
-    })(vm.networkRouterModel.ems_id);
+    if (id) {
+      getCloudNetworksByEms(id);
+      miqService.getProviderTenants(function(data) {
+        vm.available_tenants = data.resources;
+      })(id);
+    }
+    miqService.sparkleOff();
```

Comment:
Also here is no use of `Promise` and no guarantee that all data will be set before `miqService.sparkleOff()`.

Reply:
``` js
var providerTenants = miqService.getProviderTenants(function(data) {
  vm.available_tenants = data.resources;
})(id);
Promise.all([getCloudNetworksByEms(id), providerTenants])
  .then(miqService.sparkleOff)
```
But both must return API calls.

Reply:
Yes the sparkleOn/Off flow was really bad, and this is actually quite messy in general as. Since API request are already Promise I have simplified it.

Reply:
This sparkleOn/sparkleOff pair is still not doing anything and can be removed :)

Reply:
Done